### DISCUSSION TITLE
How can I generate a sidebar table of contents (TOC) based on my file?

### DISCUSSION THREAD

**Discussion by huyixi** [web-infra-dev/rspress#1373] [Q&A] [answered]
How can I generate a sidebar table of contents (TOC) based on my file?

**Comment by Timeless0911** [comment, marked as answer]
- mdx-rs: https://github.com/web-infra-dev/rspress/blob/6cc6969ef1d6bfb761605c916b21df6e3403705d/packages/core/src/node/mdx/loader.ts#L175-L182
- mdx-js: https://github.com/web-infra-dev/rspress/blob/6cc6969ef1d6bfb761605c916b21df6e3403705d/packages/core/src/node/mdx/remarkPlugins/toc.ts#L74-L85

**Reply by Timeless0911** [reply]
The built-in toc generate logic is above, I don't quite understand what you mean, is this meet your question?

**Reply by huyixi** [reply]
Yes, I already know how to handle my issue. Thank you for your help.“
